lib/upload.js

```diff
@@ -1,55 +1,102 @@
-import { hostPlatform, targetArchs } from './system.js';
-import { localPlace, remotePlace } from './places.js';
-import { log, wasReported } from './log.js';
-import { Cloud } from './cloud.js';
-import build from './build.js';
-import patchesJson from '../patches/patches.json';
-import path from 'path';
-import { verify } from './verify.js';
-import { version } from '../package.json';
-
-const cloud = new Cloud({ owner: 'zeit', repo: 'pkg-fetch' });
-
-export function dontBuild (nodeVersion, targetPlatform, targetArch) {
-  // binaries are not provided for x86 anymore
-  if (targetPlatform !== 'win' && targetArch === 'x86') return true;
-  // https://support.apple.com/en-us/HT201948
-  // don't disable macos-x86 because it is not possible
-  // to cross-compile for x86 from macos otherwise
-  const major = nodeVersion.match(/^v?(\d+)/)[1] | 0;
-  // node 0.12 does not compile on arm
-  if (/^arm/.test(targetArch) && major === 0) return true;
-  if (targetPlatform === 'freebsd' && major < 4) return true;
-  if (targetPlatform === 'alpine' &&
-      (targetArch !== 'x64' || major < 6)) return true;
-  return false;
+import fs from "fs-extra";
+import path from "path";
+
+import { hostArch, hostPlatform } from "./system.js";
+import { localPlace, remotePlace } from "./places.js";
+import { log } from "./log.js";
+import { Cloud } from "./cloud.js";
+import build from "./build.js";
+import patchesJson from "../patches/patches.json";
+import { verify } from "./verify.js";
+import { version } from "../package.json";
+import semver from 'semver';
+
+const cloud = new Cloud({ owner: 'vercel', repo: 'pkg-fetch' });
+
+/**
+ * Returns latest 3 lts versions available in patches
+ */
+function latestLts() {
```

Comment:
We do not need this method. 

Simply remove no longer supported patches instead.

Reply:
Humm I would keep them there for the ones looking for a different version, this is just for use when building and allows to keep a reference of old pathces

Reply:
No. We either support them or we don’t. And the answer is that we can’t support all old versions. 

People can check them out from Git history and support them themselves if necessary. 

Plus, such logics lead to potential inconsistency and non-determinism.